### PR TITLE
67 add eye support

### LINKED ISSUE #67: eye support

### PR DESCRIPTION
**Pull request by nicolasmery**
close #67

### CODE REVIEW

File: lib/negroku/tasks/eye.rake [hunk 1]
```diff
@@ -15,30 +15,44 @@ end
 
 namespace :eye do
 
-  task :start, [:process] do
-
+  desc "Loads eye config and starts monitoring"
+  task :load do 
+    on release_roles fetch(:eye_roles) do
+        execute "mkdir -p #{shared_path}/config/processes"
```
Comment (blackjid) [inline lib/negroku/tasks/eye.rake]: why the last folder `processes`??

File: lib/negroku/templates/tasks/eye/eye.rb.erb [hunk 2]
```diff
@@ -10,8 +10,8 @@ Eye.application '<%= fetch(:application)%>' do
   # except `env`, which merging down
 
   working_dir File.expand_path(File.join(File.dirname(__FILE__), %w[ processes ]))
```
Comment (blackjid) [inline lib/negroku/templates/tasks/eye/eye.rb.erb]: The working_dir shouldn't be `fetch(:current_path)` or something like that?

File: lib/negroku/tasks/eye.rake [hunk 3]
```diff
@@ -15,30 +15,44 @@ end
...
+        execute "cd #{current_path} && bundle exec eye load #{shared_path}/config/eye.rb"
+    end
   end
 
-  task :restart, [:process] do
-
+  desc "Starts monitoring"
```
Comment (blackjid) [inline lib/negroku/tasks/eye.rake]: This is limited to start and stop one process at a time right?? It would be nice to be able to start the complete application with all it's processes definitions...

File: lib/negroku/tasks/eye.rake [hunk 4]
```diff
@@ -15,30 +15,44 @@ end
 
 namespace :eye do
 
-  task :start, [:process] do
-
+  desc "Loads eye config and starts monitoring"
+  task :load do 
+    on release_roles fetch(:eye_roles) do
+        execute "mkdir -p #{shared_path}/config/processes"
+        execute "cd #{current_path} && bundle exec eye load #{shared_path}/config/eye.rb"
```
Comment (blackjid) [inline lib/negroku/tasks/eye.rake]: I would do something like this:
```ruby
within "#{current_path}" do
    execute :bundle, :exec, :eye, :load, "#{shared_path}/config/eye.rb"
end
```

File: lib/negroku/tasks/eye.rake [hunk 5]
```diff
@@ -69,6 +83,23 @@ namespace :negroku do
       end
     end
 
+    before "eye:setup", "eye:setup:discovery" do 
+      if was_required? 'capistrano3/unicorn'
+            watch_process(:unicorn, fetch(:unicorn_roles), {
```
Comment (blackjid) [inline lib/negroku/tasks/eye.rake]: why did you move this from the `unicorn.rake`? probably is better, I just want to know your thoughts behind the decision.. :)

File: lib/negroku/templates/tasks/eye/eye.rb.erb [hunk 6]
```diff
@@ -10,8 +10,8 @@ Eye.application '<%= fetch(:application)%>' do
...
-  stdall 'trash.log' # stdout,err logs for processes by default
-  env 'APP_ENV' => 'production' # global env for each processes
+  stdall '<%= fetch(:shared_path)%>/log/trash.log' # stdout,err logs for processes by default
+  env 'APP_ENV' => '<%= fetch(:rails_env)%>' # global env for each processes
```
Comment (blackjid) [inline lib/negroku/templates/tasks/eye/eye.rb.erb]: this probably came from the eye example, I don't think It have any sense here? right?

**Comment by blackjid** [comment]
Hey @nicolasmery thks for finishing this feature. I made a few comments in the code :)

File: lib/negroku/tasks/eye.rake (hunk 1 above)
Comment (nicolasmery) [inline lib/negroku/tasks/eye.rake]: actually, i'm not sure why but eye seems like it needs that folder (I got permission errors) for temporal files

File: lib/negroku/templates/tasks/eye/eye.rb.erb (hunk 2 above)
Comment (nicolasmery) [inline lib/negroku/templates/tasks/eye/eye.rb.erb]: I'm not sure because the current path changes on deploy. I'don't know what would happen to the eye monitor when its working dir changes on runtime. Maybe I'm thinking it wrong?

File: lib/negroku/tasks/eye.rake (hunk 5 above)
Comment (nicolasmery) [inline lib/negroku/tasks/eye.rake]: I thought that the only task responsible for monitoring should be eye. 

"eye" wouldnt like other tasks to mess around with its way of monitoring. Also it would be easier to decouple eye on the future.

File: lib/negroku/templates/tasks/eye/eye.rb.erb (hunk 2 above)
Comment (blackjid) [inline lib/negroku/templates/tasks/eye/eye.rb.erb]: maybe we could use  `fetch(:release_path)`. that one doesn't change, and as the idea is the deploy should be idempotent, we need to update the `eye.rb` on every deploy..

Comment (blackjid) [inline lib/negroku/templates/tasks/eye/eye.rb.erb]: well, now I think we should stick to `:current_path` I think the eye configuration is loaded on `load` and used at `restart`, I think changing the target for the `:current_path` symlink is not going to be a problem

**Comment by blackjid** [comment]
here is a interesting discussion with a few things that we might take into account.
https://github.com/kostya/eye/issues/95#issuecomment-65067947

**Comment by blackjid** [comment]
That discussion make me wonder if we should use eye as a global gem or as a project dependency... what do you think?

**Comment by nicolasmery** [comment]
I would use it as a global gem. After all negroku is all about conventions(as far as I understand). It makes sense to assume that eye is running in the deploy server. Also think what happens if the eye gem changes in one project in a way that negroku has to change. Then you have to update negroku for that project and then you should update all projects... Also a gobal gem probably makes easier to build a eye dashboard in the feature.

**Comment by blackjid** [comment]
i agree. :+1:
